### PR TITLE
Continued: F2C/CUDA transpilation

### PR DESCRIPTION
**Pull request by MichaelSt98**
Mostly related to nested/multi-level kernel calls:

* `__device__` prefix
* kwargs to args (also for inline calls)
* call name adaptation -> `<call/function name>_c`
* ...

### CODE REVIEW

File: loki/expression/symbols.py [hunk 1]
```diff
@@ -1418,6 +1418,42 @@ def clone(self, **kwargs):
...
+    def sort_kwarguments(self):
+        """
+        Sort and update the kwarguments/kw_parameters according to the order of the
+        arguments (``self.routine.arguments``) and return the
+        conveted clone/copy of the inline call.
+        """
+        new_kwarguments = self._sort_kwarguments()
+        return self.clone(kw_parameters=new_kwarguments)
+
+    def convert_kwargs_to_args(self):
```
Comment (reuterbal) [inline loki/expression/symbols.py:1449]: Same here: `clone_with_kwargs_as_args`?

**Review by reuterbal — CHANGES_REQUESTED** [review]
Thanks, this looks good. I would like to change the naming of the new methods for InlineCall. I got bitten by the difference in behaviour and we should make this explicit.

File: loki/transformations/transpile/fortran_c.py [hunk 2]
```diff
@@ -231,6 +230,19 @@ def transform_subroutine(self, routine, **kwargs):
             header_path = (path/c_kernel.name.lower()).with_suffix('.h')
             Sourcefile.to_file(source=self.codegen(c_kernel, header=True), path=header_path)
 
+    def convert_kwargs_to_args(self, routine, targets):
+        # calls (to subroutines)
+        for call in as_tuple(FindNodes(CallStatement).visit(routine.body)):
+            if str(call.name).lower() in as_tuple(targets):
+                call.convert_kwargs_to_args()
```
[the remark this answers is not in the document]
Comment (MichaelSt98) [reply loki/transformations/transpile/fortran_c.py:237]: That is correct!
Would it make more sense to make this an in-place update (if it is feasible and reasonable?) ?

File: loki/expression/symbols.py [hunk 3]
```diff
@@ -1418,6 +1418,42 @@ def clone(self, **kwargs):
...
+        arguments (``self.routine.arguments``)`.
+        """
+        routine = self.routine
+        assert routine is not BasicType.DEFERRED
+        kwargs = CaseInsensitiveDict(self.kwarguments)
+        r_arg_names = [arg.name for arg in routine.arguments if arg.name in kwargs]
+        new_kwarguments = tuple((arg_name, kwargs[arg_name]) for arg_name in r_arg_names)
+        return new_kwarguments
+
+    def check_kwarguments_order(self):
```
[the remark this answers is not in the document]
Comment (MichaelSt98) [reply loki/expression/symbols.py:1433]: Sure! However, that means we would have a differing API compared to the `CallStatement` IR node (although the behaviour is the same not like the convert thingy).

[the remark this answers is not in the document]
Comment (reuterbal) [reply loki/expression/symbols.py:1433]: Ah, yes, we should be consistent. Would it make sense to change it there, too?

File: loki/transformations/transpile/fortran_c.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (reuterbal) [reply loki/transformations/transpile/fortran_c.py:237]: That could be an option but I don't think we can rebuild expression nodes in-place. (Correct me if I'm wrong @mlange05)

File: loki/expression/symbols.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichaelSt98) [reply loki/expression/symbols.py:1433]: Yes, I think we can and should do that 👍

**Review by reuterbal — APPROVED** [review]
Many thanks for the changes! I think this looks ready now!

**Pull request merged**